test_localzone.py requires network access

If one keeps test_localzone.py test with no network available it fails out:

```
___________________________________________________________________________ ERROR collecting lexicon/tests/providers/test_localzone.py ____________________________________________________________________________
lexicon/tests/providers/test_localzone.py:17: in <module>
    class LocalzoneProviderTests(TestCase, IntegrationTests):
lexicon/tests/providers/test_localzone.py:23: in LocalzoneProviderTests
    filename, headers = urlretrieve(file_uri)
/usr/lib64/python2.7/urllib.py:98: in urlretrieve
    return opener.retrieve(url, filename, reporthook, data)
/usr/lib64/python2.7/urllib.py:245: in retrieve
    fp = self.open(url, data)
/usr/lib64/python2.7/urllib.py:213: in open
    return getattr(self, name)(url)
/usr/lib64/python2.7/urllib.py:443: in open_https
    h.endheaders(data)
/usr/lib64/python2.7/httplib.py:1038: in endheaders
    self._send_output(message_body)
/usr/lib64/python2.7/httplib.py:882: in _send_output
    self.send(msg)
/usr/lib64/python2.7/httplib.py:844: in send
    self.connect()
/usr/lib64/python2.7/httplib.py:1255: in connect
    HTTPConnection.connect(self)
/usr/lib64/python2.7/httplib.py:821: in connect
    self.timeout, self.source_address)
/usr/lib64/python2.7/socket.py:557: in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
E   IOError: [Errno socket error] [Errno -3] Temporary failure in name resolution
```

Could you please mark this as a network test and skip it if there is no network present?

--- a/.gitignore
+++ b/.gitignore
@@ -46,6 +46,7 @@ nosetests.xml
 coverage.xml
 *,cover
 .pytest_cache
+db.example.com
 
 # Translations
 *.mo

--- a/lexicon/tests/providers/test_localzone.py
+++ b/lexicon/tests/providers/test_localzone.py
@@ -1,28 +1,51 @@
 """Integration tests for Localzone"""
 from unittest import TestCase
 
+import os
 import pytest
 from lexicon.tests.providers.integration_tests import IntegrationTests
 
-try:
-    from urllib.request import urlretrieve
-except ImportError:
-    from urllib import urlretrieve
+FILENAME = "db.example.com"
+ZONEFILE = """
+$ORIGIN example.com.     ; designates the start of this zone file in the namespace
+$TTL 1h                  ; default expiration time of all resource records without their own TTL value
+example.com.  IN  SOA   ns.example.com. username.example.com. ( 2007120710 1d 2h 4w 1h )
+example.com.  IN  NS    ns                    ; ns.example.com is a nameserver for example.com
+example.com.  IN  NS    ns.somewhere.example. ; ns.somewhere.example is a backup nameserver for example.com
+example.com.  IN  MX    10 mail.example.com.  ; mail.example.com is the mailserver for example.com
+@             IN  MX    20 mail2.example.com. ; equivalent to above line, "@" represents zone origin
+@             IN  MX    50 mail3              ; equivalent to above line, but using a relative host name
+example.com.  IN  A     192.0.2.1             ; IPv4 address for example.com
+              IN  AAAA  2001:db8:10::1        ; IPv6 address for example.com
+ns            IN  A     192.0.2.2             ; IPv4 address for ns.example.com
+              IN  AAAA  2001:db8:10::2        ; IPv6 address for ns.example.com
+www           IN  CNAME example.com.          ; www.example.com is an alias for example.com
+wwwtest       IN  CNAME www                   ; wwwtest.example.com is another alias for www.example.com
+mail          IN  A     192.0.2.3             ; IPv4 address for mail.example.com
+mail2         IN  A     192.0.2.4             ; IPv4 address for mail2.example.com
+mail3         IN  A     192.0.2.5             ; IPv4 address for mail3.example.com
+@             IN  TXT   "v=spf1 mx ~all"      ; SPFv1 record for example.com
+"""
 
+@pytest.fixture
+def testfile(tmpdir):
+    test_dir = tmpdir.mkdir("localzone")
+    test_file = test_dir.join(FILENAME)
+    test_file.write(ZONEFILE)
+    os.chdir(str(test_dir))
 
 # Hook into testing framework by inheriting unittest.TestCase and reuse
 # the tests which *each and every* implementation of the interface must
 # pass, by inheritance from define_tests.TheTests
+@pytest.mark.usefixtures("testfile")
 class LocalzoneProviderTests(TestCase, IntegrationTests):
     """Integration tests for Localzone"""
     provider_name = "localzone"
     domain = "example.com"
-    file_uri = "https://raw.githubusercontent.com/ags-slc/localzone/master/tests/zonefiles/db.example.com"  # pylint: disable=line-too-long
-    filename, headers = urlretrieve(file_uri)
 
     def _test_parameters_overrides(self):
         options = {
-            "filename": self.filename
+            "filename": FILENAME
         }
 
         return options

--- a/lexicon/tests/providers/test_localzone.py
+++ b/lexicon/tests/providers/test_localzone.py
@@ -34,6 +34,8 @@ def testfile(tmpdir):
     test_file = test_dir.join(FILENAME)
     test_file.write(ZONEFILE)
     os.chdir(str(test_dir))
+    yield
+    os.chdir(os.path.dirname(__file__))
 
 # Hook into testing framework by inheriting unittest.TestCase and reuse
 # the tests which *each and every* implementation of the interface must

--- a/lexicon/tests/providers/test_localzone.py
+++ b/lexicon/tests/providers/test_localzone.py
@@ -1,8 +1,9 @@
 """Integration tests for Localzone"""
+import os
 from unittest import TestCase
 
-import os
 import pytest
+
 from lexicon.tests.providers.integration_tests import IntegrationTests
 
 FILENAME = "db.example.com"
@@ -27,20 +28,18 @@ mail3         IN  A     192.0.2.5             ; IPv4 address for mail3.example.c
 @             IN  TXT   "v=spf1 mx ~all"      ; SPFv1 record for example.com
 """
 
-@pytest.fixture
-def testfile(tmpdir):
-    """Create a local zone file in a temporary directory for testing."""
-    test_dir = tmpdir.mkdir("localzone")
-    test_file = test_dir.join(FILENAME)
+@pytest.fixture(scope="module", autouse=True)
+def testfile():
+    """Create a local zone file for testing."""
+    test_file = open(FILENAME, "w")
     test_file.write(ZONEFILE)
-    os.chdir(str(test_dir))
+    test_file.close()
     yield
-    os.chdir(os.path.dirname(__file__))
+    os.remove(FILENAME)
 
 # Hook into testing framework by inheriting unittest.TestCase and reuse
 # the tests which *each and every* implementation of the interface must
 # pass, by inheritance from define_tests.TheTests
-@pytest.mark.usefixtures("testfile")
 class LocalzoneProviderTests(TestCase, IntegrationTests):
     """Integration tests for Localzone"""
     provider_name = "localzone"

--- a/lexicon/tests/providers/test_localzone.py
+++ b/lexicon/tests/providers/test_localzone.py
@@ -29,6 +29,7 @@ mail3         IN  A     192.0.2.5             ; IPv4 address for mail3.example.c
 
 @pytest.fixture
 def testfile(tmpdir):
+    """Create a local zone file in a temporary directory for testing."""
     test_dir = tmpdir.mkdir("localzone")
     test_file = test_dir.join(FILENAME)
     test_file.write(ZONEFILE)

